pg_shard--1.0.sql

```diff
@@ -77,3 +77,78 @@ CREATE FUNCTION worker_copy_shard_placement(table_name text, source_node_name te
 RETURNS void
 AS 'MODULE_PATHNAME'
 LANGUAGE C STRICT;
+
+CREATE FUNCTION partition_column_to_node_string(table_oid oid)
+RETURNS text
+AS 'MODULE_PATHNAME'
+LANGUAGE C;
+
+COMMENT ON FUNCTION partition_column_to_node_string(oid)
+		IS 'return textual form of distributed table''s partition column';
+
+CREATE FUNCTION sync_table_metadata_to_citus(table_name text) RETURNS VOID
+AS $sync_table_metadata_to_citus$
+	DECLARE
+		table_relation_id CONSTANT oid NOT NULL := table_name::regclass::oid;
+		dummy_shard_length CONSTANT bigint := 0;
+	BEGIN
+		-- remove shard placements for table
```

Comment:
Rebased this branch and removed the `DELETE`-related commit, so it's like it was never there.

As far as `CASCADE` goes… it doesn't require it. This is because CitusDB calls `recordDependencyOnSingleRelExpr` and `recordDependencyOn` inside of its internal function that creates the partition entry. Once #49 is resolved `pg_shard` itself will require `CASCADE` to drop a table that's still known as distributed. Additionally, CitusDB could change its implementation to use an `INSERT`/`UPDATE` trigger so even a simple `INSERT` into its metadata tables will have dependency information recorded correctly…